cloud_governance/tag_cluster/tag_non_cluster_resources.py

```diff
@@ -0,0 +1,261 @@
+                                        else:
+                                            search_tags.extend(self.__append_input_tags())
+                                        username = self.cloudtrail.get_username_by_instance_id_and_time(
+                                            item.get('LaunchTime'),
+                                            item.get('InstanceId'),
+                                            'AWS::EC2::Instance')
+                                        break
+
+            else:
+                search_tags.extend(self.__append_input_tags())
+            if username:
+                user_tags = self.iam_client.get_user_tags(username=username)
+                search_tags.append({'Key': 'createdBy', 'Value': username})
+                search_tags.extend(user_tags)
+                tag_name = username + '-' + snapshot_id[-4:]
+                search_tags.append({'Key': 'Name', 'Value': tag_name})
+            else:
+                search_tags.append({'Key': 'Name', 'Value': f'{snapshot_id[:4]}-{self.region}-{snapshot_id[-4:]}'})
+                search_tags.extend(self.__append_input_tags())
+            search_tags.append({'Key': 'StartTime', 'Value': str(snapshot.get('StartTime'))})
+            snapshot_tags = self.__get_tags_of_resources(tags=snapshot.get('Tags'), search_tags=search_tags)
+            if snapshot_tags:
+                if self.dry_run == 'no':
+                    self.ec2_client.create_tags(Resources=[snapshot_id], Tags=snapshot_tags)
+                    logger.info(
+                        f'added tags to snapshots: {snapshot_id} total: {len(snapshot_tags)} tags: {snapshot_tags}')
+                snapshot_ids.append(snapshot_id)
+        logger.info(f'non_cluster_snapshot count: {len(sorted(snapshot_ids))} {sorted(snapshot_ids)}')
+
+    def update_ami(self, images: list):
+        """
+        This method update the tags of non-cluster Amazon Machine Images
+        @param images:
+        @return:
+        """
+        image_ids = []
+        for image in images:
+            image_id = image.get('ImageId')
+            start_time = datetime.fromisoformat(image.get('CreationDate')[:-1] + '+00:00')
+            username = self.cloudtrail.get_username_by_instance_id_and_time(start_time=start_time,
+                                                                            resource_id=image_id,
+                                                                            resource_type='AWS::EC2::Ami')
+            search_tags = []
+            search_tags.extend(self.__append_input_tags())
+            if username:
+                user_tags = self.iam_client.get_user_tags(username=username)
+                tag_name = username + '-' + image_id[-4:]
+                search_tags.append({'Key': 'createdBy', 'Value': username})
+                search_tags.extend(user_tags)
+            else:
+                tag_name = image_id[:3] + '-' + self.region + '-' + image_id[-4:]
```

Comment:
Please change it to format instead of +

Reply:
tag_name = f'{image_id[:3]} - {self.region} - {image_id[-4:]}'
Remove extra space